posthog/clickhouse/client/execute.py

```diff
@@ -40,37 +40,46 @@
 
 @lru_cache(maxsize=1)
 def default_settings() -> Dict:
-    from posthog.version_requirement import ServiceVersionRequirement
-
     # On CH 22.3 we need to disable optimize_move_to_prewhere due to a bug. This is verified fixed on 22.8 (LTS),
     # so we only disable on versions below that.
     # This is calculated once per deploy
-    clickhouse_at_least_228, _ = ServiceVersionRequirement(
-        service="clickhouse", supported_version=">=22.8.0"
-    ).is_service_in_accepted_version()
-    if clickhouse_at_least_228:
+    if clickhouse_at_least_228():
         return {}
     else:
         return {"optimize_move_to_prewhere": 0}
 
 
+@lru_cache(maxsize=1)
+def clickhouse_at_least_228() -> bool:
+    from posthog.version_requirement import ServiceVersionRequirement
+
+    is_ch_version_228_or_above, _ = ServiceVersionRequirement(
+        service="clickhouse", supported_version=">=22.8.0"
+    ).is_service_in_accepted_version()
+
+    return is_ch_version_228_or_above
+
+
 def extra_settings(query_id: Optional[str], tags: Dict[str, Any]) -> Dict[str, Any]:
     if tags.get("kind") != "celery":
         return {}
 
+    # The `default` option for join_algorithm was introduced with CH 22.8
+    default_join_algorithm = "default" if clickhouse_at_least_228() else "direct,hash"
+
     join_algorithm = (
         posthoganalytics.get_feature_flag(
             "join-algorithm",
             str(query_id),
             only_evaluate_locally=True,
             send_feature_flag_events=False,
         )
-        or "default"
+        or default_join_algorithm
```

Comment:
Problem: We could return an invalid join_algorithm as as a result of this under 228.

I think we should return {} if not clickhouse_at_least_228

Reply:
Will update